sesame/src/main/scala/org/w3/banana/sesame/SesameTransactor.scala

```diff
@@ -0,0 +1,28 @@
+package org.w3.banana.sesame
+
+import org.openrdf.repository.RepositoryConnection
+import org.w3.banana.{Transactor, SparqlUpdate, RDFStore}
+
+import scala.util.{Failure, Try}
+
+trait SesameTransactor extends Transactor[Sesame,RepositoryConnection] {
+
+  def r[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
+    val result = body
+    conn.close()
+    result
+  }
+
+  def rw[T](conn: RepositoryConnection, body: => T): Try[T] = Try{
```

Comment:
I do not think that it is fully possible to have same semantics of transactions for _all_ implementation. The problem is that even the same database can behave differently with different configurations. For instance in bigdata developer can choose between many_readers and one_writer mode (that is the default and recommended one) and many_readers/many_writers mode. There are also similar configurations for other aspects.  I believe the situation is similar for optimistic/pessimistic transactions: in some database you can switch between them in config, in some one is preffered.